koku/api/report/aws/openshift/query_handler.py

```diff
@@ -61,9 +61,20 @@ def execute_query(self):  # noqa: C901
                     account_alias=Coalesce(F(self._mapper.provider_map.get("alias")), "usage_account_id")
                 )
 
-            if self._limit:
-                rank_order = getattr(F(self.order_field), self.order_direction)()
-                rank_by_total = Window(expression=RowNumber(), partition_by=F("date"), order_by=rank_order)
+            # Set default delta if one was not passed in.
+            if self.order_field == "delta":
+                if self._delta is None:
```

Comment:
Do we know why self._delta isn't being set when we are ordering by delta? Could this be a bug in the QueryParams class?

Reply:
@adberglund So, I had not considered that this might be a bug in the QueryParameters class. The `self._delta` is set by passing it into the url. So you will notice in my tests I use two different urls to test this:
```
"?filter[limit]=5&filter[offset]=0&order_by[delta]=asc",
 "?filter[limit]=5&filter[offset]=0&order_by[delta]=asc&delta=usage",
```
When the `&delta=usage` is **not** added to the url the `self._delta` is None. However, when it is added on it it becomes the value of whatever you set there. 

My solution to the problem was for when delta is none, I check the provider map for the `delta_key`, and grab the first key in that dictionary. However, that is just one possible solution, another way I could have handled this was to change the serializer to require `&delta=x` to be passed in whenever someone does `&order_by[delta]=asc`.

Reply:
@myersCody  - I think your second solution is probably the better one. This is more a validation issue than anything else. If `order_by[delta]` is used, then `delta=X` is a required parameter. So, it can't be `None` and the serializer should raise a `ValidationError` in that case.

Reply:
@blentz That works for me too! I will try to get this update in today. I will ping ya for another review on slack when I do.